plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergGlueCatalogAccessOperations.java

```diff
@@ -0,0 +1,502 @@
+                            .addCopies(GET_TABLE, 1)
+                            .build());
+
+            // select from $partitions
+            assertGlueMetastoreApiInvocations("SELECT * FROM \"test_select_snapshots$partitions\"",
+                    ImmutableMultiset.builder()
+                            .addCopies(GET_TABLE, 1)
+                            .build());
+
+            // select from $files
+            assertGlueMetastoreApiInvocations("SELECT * FROM \"test_select_snapshots$files\"",
+                    ImmutableMultiset.builder()
+                            .addCopies(GET_TABLE, 1)
+                            .build());
+
+            // select from $properties
+            assertGlueMetastoreApiInvocations("SELECT * FROM \"test_select_snapshots$properties\"",
+                    ImmutableMultiset.builder()
+                            .addCopies(GET_TABLE, 1)
+                            .build());
+
+            // This test should get updated if a new system table is added.
+            assertThat(TableType.values())
+                    .containsExactly(DATA, HISTORY, SNAPSHOTS, MANIFESTS, PARTITIONS, FILES, PROPERTIES);
+        }
+        finally {
+            getQueryRunner().execute("DROP TABLE IF EXISTS test_select_snapshots");
+        }
+    }
+
+    private void assertGlueMetastoreApiInvocations(@Language("SQL") String query, Multiset<?> expectedInvocations)
+    {
+        Map<GlueMetastoreMethod, Integer> startingCounts = Arrays.stream(GlueMetastoreMethod.values())
+                .collect(Collectors.toMap(
+                                Function.identity(),
+                                method -> (int) method.getStatFrom(catalogStats).getTime().getAllTime().getCount() +
+                                        (int) method.getStatFrom(tableOperationStats).getTime().getAllTime().getCount()
+                        )
+                );
+
+        getQueryRunner().execute(query);
+        Map<GlueMetastoreMethod, Integer> endingCounts = Arrays.stream(GlueMetastoreMethod.values())
+                .collect(Collectors.toMap(
+                                Function.identity(),
+                                method -> (int) method.getStatFrom(catalogStats).getTime().getAllTime().getCount() +
+                                        (int) method.getStatFrom(tableOperationStats).getTime().getAllTime().getCount()
+                        )
+                );
+
+        Map<GlueMetastoreMethod, Integer> deltas = Arrays.stream(GlueMetastoreMethod.values())
+                .collect(Collectors.toMap(Function.identity(), method -> endingCounts.get(method) - startingCounts.get(method)));
+        ImmutableMultiset.Builder<GlueMetastoreMethod> builder = ImmutableMultiset.builder();
+        deltas.entrySet().stream().filter(entry -> entry.getValue() > 0).forEach(entry -> builder.setCount(entry.getKey(), entry.getValue()));
+        Multiset<GlueMetastoreMethod> actualInvocations = builder.build();
```

Comment:
Do this at once.

Reply:
Im a little unclear here. Do you want me to try to put this all in one streaming line? I wasn't able to figure out how without a stream.fold() function available. Is there a collector that could handled the predefined counts of objects?